Cython/Compiler/PyrexTypes.py

```diff
@@ -4657,7 +4657,8 @@ def specialize(self, values):
 def c_tuple_type(components):
     components = tuple(components)
     if any(c.is_fused for c in components):
-        cname = "<dummy fused ctuple>"  # should never end up in code
+        # should never end up in code but should be unique
+        cname = "<dummy fused ctuple %s>" % repr(components)
```

Comment:
`"%s" % repr(obj)` is better spelled `"%r" % obj`, but let's use an f-string regardless.
```suggestion
        cname = f"<dummy fused ctuple {components:r}>"
```

Reply:
> `"%s" % repr(obj)` is better spelled `"%r" % obj`

It wasn't quite the same. Because `obj` is a sequence, it requires a `%` for each element. I suspect `"%r" % (obj, )` would work though